internal/etag/etag.go

```diff
@@ -143,15 +143,30 @@ func (e ETag) String() string {
 
 // IsEncrypted reports whether the ETag is encrypted.
 func (e ETag) IsEncrypted() bool {
-	return len(e) > 16 && !bytes.ContainsRune(e, '-')
+	// An encrypted ETag must be at least 32 bytes long.
+	// It contains the encrypted ETag value + an authentication
+	// code generated by the AEAD cipher.
+	//
+	// Here is an incorrect implementation of IsEncrypted:
+	//
+	//   return len(e) > 16 && !bytes.ContainsRune(e, '-')
+	//
+	// An encrypted ETag may contain some random bytes - e.g.
+	// and nonce value. This nonce value may contain a '-'
+	// just by its nature of being randomly generated.
+	// The above implementation would incorrectly consider
+	// such an ETag (with a nonce value containing a '-')
+	// as non-encrypted.
+
+	return len(e) >= 32 // We consider all ETags longer than 32 bytes as encrypted
```

Comment:
@aead, should this be len(e) > 32

Reply:
Why `>` based on the code doc an encrypted ETag consists of 16 bytes for its actual value + 16 bytes for the authentication tag. That is `32`.

In general, ETags encrypted using our `sio` package also contain a random value and some cipher / encryption format version information. So they are at least `16 + 16 + 1 + 1 + 12` bytes long. However, that seems like a implementation detail.
If the encryption mechanism for ETags changes, this code may again become buggy.

A secure encryption of some value will always add at least a 16 byte checksum. So the minium size of encrypted ETag is its plaintext length + 16. Since a ETag must be at least 16 bytes long, `32` is the minimum for an encrypted ETag.

Reply:
Also, a multipart ETag cannot be `32` bytes long. Then it would not be valid. 

Note that we check the length of the binary ETag, not is hex-encoded length